controllers/kustomization_controller.go

```diff
@@ -507,7 +527,26 @@ func (r *KustomizationReconciler) validate(kustomization kustomizev1.Kustomizati
 	return nil
 }
 
-func (r *KustomizationReconciler) getKubeConfig(kustomization kustomizev1.Kustomization, dirPath string) (string, error) {
+func (r *KustomizationReconciler) getKubeConfigPath(kustomization kustomizev1.Kustomization, dirPath string) (string, error) {
```

Comment:
With this we are reading the secret twice on each reconcile run. I think we should change it to:
```
func (r *KustomizationReconciler) witeKubeConfig(data [][]byte, dirPath string) (string, error)
```